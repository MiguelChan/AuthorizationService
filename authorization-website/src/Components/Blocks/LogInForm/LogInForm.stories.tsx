import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { LogInForm, LogInFormProps } from './LogInForm';

export default {
  title: 'Components/Blocks/LogInForm',
  component: LogInForm,
  argTypes: {
    onLogIn: {
      defaultValue: 'onLogIn',
      description: 'onLogIn',
      name: 'onLogIn',
    }
  }
} as Meta;

const Template: Story<LogInFormProps> = (args) => <LogInForm {...args} />;

export const Primary = Template.bind({});