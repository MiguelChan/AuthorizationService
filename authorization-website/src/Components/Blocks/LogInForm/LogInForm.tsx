import { 
  Grid,
  IconButton, 
  InputAdornment,
  TextField,
  Typography,
  useTheme,
} from '@mui/material';
import React from 'react';
import {
  useFormik,
} from 'formik';
import * as yup from 'yup';
import { 
  Visibility, 
  VisibilityOff,
} from '@mui/icons-material';
import { LoadingButton } from '@mui/lab';

const validationSchema = yup.object({
  email: yup
    .string()
    .email()
    .required('Email is required'),
  password: yup
    .string()
    .min(8, 'Password must be 8 characters minimum')
    .max(20, 'Password cant be longer than 20 characters')
    .required('Password is required'),
});

export interface LogInFormProps {
  onLogIn: (username: string, password: string) => void;
  isLoading: boolean;
}

export const LogInForm: React.FunctionComponent<LogInFormProps> = ({
  onLogIn,
  isLoading,
}) => {

  const currentTheme = useTheme();
  const [hidePassword, setHidePassword] = React.useState<boolean>(true);
  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
    },
    validationSchema,
    onSubmit: (values) => {
      onLogIn(values.email, values.password);
    },
  });

  const getInputAdornment = (): React.ReactElement => (
    <InputAdornment position='end'>
      <IconButton 
        aria-label='toggle password visibility'
        onClick={(): void => setHidePassword(!hidePassword)}
        edge='end'
      >
        {hidePassword ? <Visibility /> : <VisibilityOff />}
      </IconButton>
    </InputAdornment>
  );

  const centeredStyle: React.CSSProperties = {
    flexGrow: '1',
    textAlign: 'center',
  };

  return (
    <Grid container spacing={2} marginTop={currentTheme.spacing(1)}>
      <Grid item xs={12}>
        <Typography 
          variant='body1'
          sx={centeredStyle}
        >
          Log into the Auth-Hub
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <TextField 
          fullWidth
          id='email'
          name='email'
          label='Email'
          value={formik.values.email}
          onChange={formik.handleChange}
          error={formik.touched.email && Boolean(formik.errors.email)}
          helperText={formik.touched.email && formik.errors.email}
        />
      </Grid>
      <Grid item xs={12}>
        <TextField
          fullWidth
          id='password'
          name='password'
          label='Password'
          type={hidePassword ? 'password' : 'text'}
          value={formik.values.password}
          onChange={formik.handleChange}
          error={formik.touched.password && Boolean(formik.errors.password)}
          helperText={formik.touched.password && formik.errors.password}
          InputProps={{
            endAdornment: getInputAdornment(),
          }}
        />
      </Grid>
      <Grid item xs='auto' sx={centeredStyle}>
        <LoadingButton 
          variant='contained' 
          onClick={formik.submitForm}
          loading={isLoading}
        >
          Log In
        </LoadingButton>
      </Grid>
    </Grid>
  );
};