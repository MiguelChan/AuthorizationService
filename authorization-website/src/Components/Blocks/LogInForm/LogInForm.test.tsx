import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { LogInForm } from './LogInForm';

describe('LogInForm', () => {

  const mockOnLogInFn = jest.fn();

  afterEach(() => {
    mockOnLogInFn.mockClear();
  });

  it('Should display the LogInForm', () => {
    const renderResult = setupComponent();
    expect(renderResult.container).toMatchSnapshot();
  });

  const setupComponent = (): RenderResult => {
    return render(
      <LogInForm 
        onLogIn={mockOnLogInFn} 
        isLoading={false}
      />
      );
  };

});