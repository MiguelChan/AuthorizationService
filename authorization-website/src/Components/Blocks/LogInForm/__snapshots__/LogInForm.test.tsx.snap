// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm Should display the LogInForm 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-5sfu98-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1qvkqva-MuiTypography-root"
      >
        Log into the Auth-Hub
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="email"
          id="email-label"
        >
          Email
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="email"
            name="email"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1ntzz89-MuiGrid-root"
    >
      <button
        class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        tabindex="0"
        type="button"
      >
        Log In
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
